soc/nordic/nrf54h/pinctrl.c

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2025 Nordic Semiconductor ASA.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <nrf/gpd.h>
```

Comment:
something seems wrong with the includes here, this file should include `zephyr/drivers/pinctrl/nrf_pinctrl.h` (a file which does not exist yet, but it should, the hooks are device driver specific, not soc specific)